src/main/java/org/micks/champmaker/auth/AuthService.java

```diff
@@ -19,4 +22,18 @@ public void createUser(CreateUserRequest createUserRequest) {
         String encryptedPassword = passwordService.encryptPassword(createUserRequest.getPassword());
         userPersistApi.storeUser(createUserRequest.getUsername(), encryptedPassword);
     }
+
+    public void loginUser(LoginRequest loginRequest) {
+        List<User> users = userPersistApi.readFile();
+        String loginRequestUsername = loginRequest.getUsername();
+        String loginRequestPassword = loginRequest.getPassword();
+        Optional<User> first = users.stream()
+                .filter(user -> user.getUsername().equals(loginRequestUsername))
+                .findFirst();
+        User foundUser = first.orElseThrow();
+        String encryptPassword = passwordService.encryptPassword(loginRequestPassword);
+        if (!foundUser.getEncryptedPassword().equals(encryptPassword)) {
+            throw new RuntimeException("Login failed");
```

Comment:
This should be exception more like custom checked exception mapped to 404 status.